installer/engine_installer/engine_installer.go

```diff
@@ -14,8 +14,168 @@
 
 package main
 
-import "fmt"
+import (
+	"context"
+	"fmt"
+	"html/template"
+	"net/url"
+	"strings"
 
-func main() {
-	fmt.Println("hello world")
+	"github.com/vmware/vic/lib/install/data"
+	"github.com/vmware/vic/lib/install/validate"
+)
+
+// EngineInstallerConfigOptions contains resource options for selection by user in exec.html
+type EngineInstallerConfigOptions struct {
+	Networks      []string
+	Datastores    []string
+	ResourcePools []string
+}
+
+// EngineInstaller contains all options to be used in the vic-machine create command
+type EngineInstaller struct {
+	BridgeNetwork   string `json:"bridge-net"`
+	PublicNetwork   string `json:"public-net"`
+	ImageStore      string `json:"img-store"`
+	ComputeResource string `json:"compute"`
+	Target          string `json:"target"`
+	User            string `json:"user"`
+	Password        string `json:"password"`
+	Name            string `json:"name"`
+	Thumbprint      string `json:"thumbprint"`
+	CreateCommand   string
+}
+
+// AuthHTML holds the invalid login variable
+type AuthHTML struct {
+	InvalidLogin bool
+}
+
+// ExecHTMLOptions contains fields for html templating in exec.html
+type ExecHTMLOptions struct {
+	BridgeNetwork   template.HTML
+	PublicNetwork   template.HTML
+	ImageStore      template.HTML
+	ComputeResource template.HTML
+	Target          string
+	User            string
+	Password        string
+	Name            string
+	Thumbprint      string
+	CreateCommand   string
+}
+
+var (
+	validator *validate.Validator
+)
+
+// NewEngineInstaller returns a new EngineInstaller struct with empty parameters
```

Comment:
Being pedantic, but this comment is invalid - the struct doesn't have empty parameters per-se since the `Name` field is populated. I'd recommend just nuking the comment since the func is self-explanatory.